scripts/setup/docker.sh

```diff
@@ -0,0 +1,220 @@
+#
+print_warning() {
+    echo -e "${YELLOW}[WARN] ${1}${NC}"
+}
+
+#
+# DESCRIPTION
+#   Gets the rust version used in the project.
+#
+# RETURNS
+#   - On success, a string containing the Rust version (e.g., "nightly-2025-03-28").
+#   - On failure, an empty string.
+#
+# USAGE EXAMPLE
+#   RUST_VERSION=$(get_rust_version)
+#
+get_rust_version() {
+    local rust_toolchain_file
+    rust_toolchain_file=$(mktemp)
+
+    # Get rust-toolchain file from the default branch.
+    git -C "${REPO_ROOT_DIR}" show origin/${DEFAULT_BRANCH_NAME}:rust-toolchain > "${rust_toolchain_file}" 2>/dev/null || {
+        rm -f "${rust_toolchain_file}"
+        print_warning "Could not find 'rust-toolchain' file in the repository."
+        echo ""
+        return
+    }
+
+    # Extract the line containing the Rust channel.
+    local channel_line
+    channel_line=$(grep -E '^[[:space:]]*channel[[:space:]]*=' "${rust_toolchain_file}" | head -n 1)
+
+
+    # If the channel line is not empty, assume 'rust-toolchain' is in TOML format.
+    # Otherwise, assume it is in plain text format (single line with version).
+    local version=""
+    if [ -n "${channel_line}" ]; then
+        # Try to extract the channel from TOML format.
+        # Get the value after '=' and remove spaces and quotes.
+        version=$(echo "${channel_line}" | cut -d'=' -f2 | tr -d '[:space:]' | tr -d '"')
+    else
+        # Fallback: assume plain text format (single line with version).
+        version=$(head -n 1 "${rust_toolchain_file}" | tr -d '[:space:]')
+    fi
+
+    rm -f "${rust_toolchain_file}"
+
+    # Check if the version string is empty.
+    if [ -z "${version}" ]; then
+        print_error "Failed to extract Rust version from 'rust-toolchain' file."
+        echo ""
+        return
+    fi
+
+    echo "${version}"
+}
+
+#==================================================================================================
+# Main Script
+#==================================================================================================
+#
+# DESCRIPTION
+#   Builds the Docker image for the Nanvix toolchain.
+#
+main() {
+    print_message "Building Docker image for Nanvix toolchain..."
+
+    # Sanity check that we are running inside a git repository.
+    if ! git rev-parse --is-inside-work-tree &> /dev/null; then
+        print_error "This script must be run inside a Git repository."
+        exit 1
+    fi
+
+    # Sanity check that branch name was set correctly.
+    if [ -z "${DEFAULT_BRANCH_NAME}" ]; then
+        print_error "Could not determine the default branch name."
+        exit 1
+    fi
+
+    # Sanity check that repository directory was set correctly.
+    if [ ! -d "${REPO_ROOT_DIR}" ]; then
+        print_error "Could not determine the repository root directory."
+        exit 1
+    fi
+
+    # Sanity check that docker is installed.
+    if ! command -v docker &> /dev/null; then
+        print_error "Docker is not installed."
+        exit 1
+    fi
+
```

Comment:
Should we add a check to see if `docker` is in `groups`? Otherwise we would need `sudo` to run `docker build`.

Maybe we should add a note in the docs somewhere specifying that we assume you can run `docker` without `sudo`, too.

Reply:
Yes, we should. Good catch!